langchain/src/vectorstores/supabase.ts

```diff
@@ -0,0 +1,107 @@
+import { SupabaseClient } from '@supabase/supabase-js';
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+
+type SupabaseMetadata = Record<string, any>;
+
+export type SearchEmbeddingsResponse = {
+    id: number
+    prompt: string
+    completion: string
+    similarity: number
+    token_count: number
+  }
+
+  export type SearchEmbeddingsParams = {
+    query_embedding: number[]
+    match_threshold: number // float
+    match_count: number // int
+  }
+  
+  
+export class PGVectorStore extends VectorStore {
+  textKey: string;
+
+  supabaseClient: SupabaseClient;
+
+  constructor(
+    supabaseClient: SupabaseClient,
+    embeddings: Embeddings,
+    textKey = "text"
+  ) {
+    super(embeddings);
+
+    this.supabaseClient = supabaseClient;
+    this.embeddings = embeddings;
+    this.textKey = textKey;
+  }
+
+  async addDocuments(_documents: Document[], _ids?: string[]): Promise<void> {
+    return new Promise((resolve) => {
+        resolve();
+      });
+  }
+
+  async addVectors(
+    _vectors: number[][],
+    _documents: Document[],
+    _ids?: string[]
+  ): Promise<void> {
+    return new Promise((resolve) => {
+        resolve();
+      });
+  }
+
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number
+  ): Promise<[Document, number][]> {
+    const matchDocumentsParams: SearchEmbeddingsParams = {
+      query_embedding: query,
+      match_threshold: k,
```

Comment:
Is this right? K is meant to be the number of similar documents we want returned

Reply:
oh, is there a similarity score cut off?